example/advection/parthenon_app_inputs.cpp

```diff
@@ -45,39 +46,40 @@ void ProblemGenerator(MeshBlock *pmb, ParameterInput *pin) {
   const auto &sin_a3 = pkg->Param<Real>("sin_a3");
   const auto &profile = pkg->Param<std::string>("profile");
 
-  auto q_h = q.GetHostMirror();
-
   auto cellbounds = pmb->cellbounds;
-
   IndexRange ib = cellbounds.GetBoundsI(IndexDomain::entire);
   IndexRange jb = cellbounds.GetBoundsJ(IndexDomain::entire);
   IndexRange kb = cellbounds.GetBoundsK(IndexDomain::entire);
 
   auto coords = pmb->coords;
-
-  for (int k = kb.s; k <= kb.e; k++) {
-    for (int j = jb.s; j <= jb.e; j++) {
-      for (int i = ib.s; i <= ib.e; i++) {
-        if (profile.compare("wave") == 0) {
+  auto q = data->PackVariables(std::vector<MetadataFlag>{Metadata::Independent});
+  const auto num_vars = q.GetDim(4);
+
+  int profile_type;
+  if (profile == "wave") profile_type = 0;
+  if (profile == "smooth_gaussian") profile_type = 1;
+  if (profile == "hard_sphere") profile_type = 2;
```

Comment:
-> enum class 
Feel free to add it to the list of items to be updated with a more complete test.

Reply:
Added